### ISSUE TITLE
Some minor  issues

### ISSUE THREAD

**Issue by RobTillaart** [TilenS6/ArduinoSerialDraw-Library#4]
Interesting concept library, so some remarks to improve it

*Arduino.h*

You include "Arduino.h" in the SerialDraw.h file  and again in the SerialDraw.cpp. 
As the latter includes SerialDraw.h it automatically gets Arduino.h

----

*Prevention of double include*

```cpp
#ifndef SerialDraw_h
#define SerialDraw_h

#endif
```
is often replaced by oneliner at the top of the .h file.  Compiler does the administration for you.
```cpp
#pragma once
```
----
*GotIt*

nowhere used outside waitForResponse ==> local variable.

----
*waitForResponse* can be simpler. + use **yield()* 
```cpp
void waitForResponse()
{
  uint32_t start = millis();
  while (millis() - start < 100)
  {
    yield();  // needed for ESP32 / 8266 support (has to do with wifi in background something)
    if (Serial.available > 0)
    {
      Serial.readBytes(in, 1);  // do not read more as otherwise the array will overflow.
       if (in[0] == 'k') break;
    }
  }
}
````
----
*do not stringify a string*
```cpp
void ASD::pixel(int x, int y, String color) {
  String _color = stringToColor(color);
  Serial.println("p," + String(x) + "," + String(y) + "," + String(_color));  // why stringify color as it is a string?
  waitForResponse();
}
```
should be
```cpp
void ASD::pixel(int x, int y, String color)
{
  Serial.println("p," + String(x) + "," + String(y) + "," + stringToColor(color));  // function returns a string so 
  waitForResponse();
}
```
pattern is seen in more functions.

The usage of String() class is discussed a lot as it uses dynamic memory. 
In short it can make "Swiss cheese" of your heap, causing a sudden crash after some undefined time. 
A quick look seems that you only use Strings quite clean (maybe too much but that is a matter of style)

**Comment by TilenS6** [comment]
@RobTillaart Thank you so much and we appreciate that! I was coding this library and I was making library for the first time so it got me there. Issues were fixed and will be applied soon.
Thanks!
